lib/features/push_notification/data/datasource/web_socket_datasource.dart

```diff
@@ -0,0 +1,6 @@
+import 'package:jmap_dart_client/jmap/account_id.dart';
+import 'package:jmap_dart_client/jmap/core/session/session.dart';
+
+abstract class WebSocketDatasource {
+  Stream<dynamic> getWebSocketChannel(Session session, AccountId accountId);
```

Comment:
Why `dynamic`? Don't we have a specific object from the server?

Reply:
No we don't. According to the [specs](https://www.rfc-editor.org/rfc/rfc8887.html#name-examples), server web socket response takes many forms.